### ISSUE TITLE
batch upsert/update/patch but with errors

### ISSUE THREAD

**Issue by thoughtworks-tcaceres** [reduxjs/redux-toolkit#4777]
Hello,
I know of the `upsertQueryData`, `patchQueryData`, and `updateQueryData`, but it looks like these 3 ways are for handling successful data only.

Currently we have an api call - lets call it `getProduct`, and we may need to do 100 calls. So what we will be making soon is a `getProducts` api  call.
Once we get the data from this batch api call, we want to update each cache individually using upsertQueryData.
However, depending on the user and their permissions, they may not have access to certain data which (in our UI) should show an error message when they navigate to that page.
In this case, our api request also sends a "status" (http status) field as part of the batch request which we were hoping we could leverage.
However, it looks like the 3 methods mentioned above only work with the 'happy path' (data) and not with errors.
Is there any current way to get around this?

tl;dr - want to not only upsert Data, but also the errors that may be returned (http status code, etc) from an api call that is designed for batching requests.

**Comment by markerikson** [comment]
@thoughtworks-tcaceres no, there's not currently a way to intentionally insert or mark entries as errored.

Also, note that you'll almost definitely want to use the new `upsertQueryEntries` method for batch-inserting values into cache entries, as it's significantly more efficient and faster than `upsertQueryData`:

- https://redux-toolkit.js.org/rtk-query/api/created-api/api-slice-utils#upsertqueryentries

**Comment by thoughtworks-tcaceres** [comment]
> [@thoughtworks-tcaceres](https://github.com/thoughtworks-tcaceres) no, there's not currently a way to intentionally insert or mark entries as errored.
> 
> Also, note that you'll almost definitely want to use the new `upsertQueryEntries` method for batch-inserting values into cache entries, as it's significantly more efficient and faster than `upsertQueryData`:
> 
> * https://redux-toolkit.js.org/rtk-query/api/created-api/api-slice-utils#upsertqueryentries

Thanks for the swift response! Much appreciated.
That's unfortunate that there's no way to mark an errored entry for batch requests. I will let my team know.
However, great news for the new `upsertQueryData` -- sounds amazing! 😀

**Comment by thoughtworks-tcaceres** [comment]
Is this something you think would be worth adding to feature requests?
I'll make one if anyone thinks this would be valuable?
